src/commands/scanner/rule/remove.ts

```diff
@@ -80,7 +80,7 @@ export default class Remove extends ScannerCommand {
 		if (!this.flags.force) {
 			// Step 6a: We'll want to create filter criteria.
 			const filters: RuleFilter[] = [];
-			filters.push(new RuleFilter(FilterType.SOURCEPACKAGE, deletablePaths));
+			filters.push(new SourcePackageFilter(deletablePaths));
```

Comment:
This change converts the RuleFilters to strongly typed classes, lots of similar changes in other files.